test/components/submission/download-button.spec.js

```diff
@@ -23,19 +23,27 @@ describe('SubmissionDownloadButton', () => {
   describe('text', () => {
     it('shows the correct text if the submissions are not filtered', () => {
       testData.extendedForms.createPast(1, { submissions: 2 });
-      mountComponent().text().should.equal('Download 2 Submissions…');
+      mountComponent().find('.btn-primary').text().should.equal('Download');
     });
 
     describe('submissions are filtered', () => {
-      it('shows correct text while first chunk of submissions is loading', () => {
+      it('show the dropdown menu', () => {
         testData.extendedForms.createPast(1, { submissions: 2 });
         const component = mountComponent({
           props: { filtered: true }
         });
-        component.text().should.equal('Download matching Submissions…');
+        component.find('.btn-primary').attributes()['data-toggle'].should.be.eql('dropdown');
       });
 
-      it('shows correct text after first chunk of submissions has loaded', () => {
+      it('shows correct text while first chunk of submissions is loading for the first button', () => {
+        testData.extendedForms.createPast(1, { submissions: 2 });
+        const component = mountComponent({
+          props: { filtered: true }
+        });
+        component.find('li:nth-of-type(1)').text().should.equal('Download all Submissions matching the filter');
+      });
+
+      it('shows correct text after first chunk of submissions has loaded for the first button', () => {
```

Comment:
was confused by "first button" and "second button" in test title. you mean the 1st and 2nd dropdown option (shown only when the submissions are filtered)?